s3.go

```diff
@@ -75,6 +83,7 @@ func NewS3Datastore(conf Config) (*S3Bucket, error) {
 	}
 
 	awsConfig.WithCredentials(creds)
+	awsConfig.CredentialsChainVerboseErrors = aws.Bool(true)
```

Comment:
?

Reply:
It helps the user of go-ds-s3 that the errs from each provider gets passed down, as it can otherwise take more time debugging to find out the details when dealing with multiple potential error sources. https://github.com/aws/aws-sdk-go/blob/master/aws/config.go#L34 and https://github.com/aws/aws-sdk-go/blob/master/aws/credentials/chain_provider.go#L86